tests/ui/object-lifetime/object-lifetime-default-gat-resolved.rs

```diff
@@ -0,0 +1,33 @@
+// Check that we correctly deduce object lifetime defaults inside resolved GAT *paths*.
+// issue: <https://github.com/rust-lang/rust/issues/115379>
+//@ check-pass
+
+mod own { // the object lifetime default comes from the own generics
+    trait Outer {
+        type Ty<'a, T: ?Sized + 'a>;
+    }
+    impl Outer for () {
+        type Ty<'a, T: ?Sized + 'a> = &'a T;
+    }
+    trait Inner {}
+
+    fn f<'r>(x: <() as Outer>::Ty<'r, dyn Inner + 'r>) { g(x) }
+    // We deduce `dyn Inner + 'r` from bound `'a` on ty param `T` of assoc ty `Ty`.
+    fn g<'r>(_: <() as Outer>::Ty<'r, dyn Inner>) {}
+}
+
+mod parent { // the object lifetime default comes from the parent generics
```

Comment:
Also add a test with both self and parent lifetimes

Reply:
Do you mean one where the assoc ty is `type AssocTy<'own, T: ?Sized + 'own + 'parent>;` or `type AssocTy<'own, T: ?Sized + 'own, U: ?Sized + 'parent>;` or something else entirely?

Reply:
I had in mind just `type AssocTy<'own, T: ?Sized + 'own>` (basically identical to the first example, but showing that the extra lifetime doesn't throw it off), and ideally some sort of failure tests of cases that are using the wrong lifetime.